modules/es-extensions/publisher/modules/greg-publisher-api.js

```diff
@@ -705,5 +709,147 @@ var gregAPI = {};
             properties = collection.getProperties();
         }
         return properties;
-    }
+    };
+
+    /***
+     * Calls the registry util class to get the role permissions for a particular resource
+     * @param am
+     * @param assetId
+     * @returns {{}}
+     */
+    gregAPI.permissions.list = function (am, assetId) {
+        var registryPath = am.get(assetId).path;
+        var userRegistry = am.registry;
+        var registry = userRegistry.registry;
+        var PermissionUtil = Packages.org.wso2.carbon.registry.resource.services.utils.PermissionUtil;
+        var results = {};
+        var result = [];
+        var permissionCheck;
+        var resource = registry.get(registryPath);
+        try {
+            if(resource){
+                permissionCheck = resource.getProperty(REGISTRY_PERMISSION_CHECK);
+            }
+
+            var permissionsBean = PermissionUtil.getPermissions(userRegistry.registry, registryPath);
+            if (permissionsBean) {
+                var permissions = permissionsBean.getRolePermissions();
+                var authorizedRoles = [];
+
+                for (var i = 0; i < permissions.length; i++) {
+                    var permissionOptions = {};
+                    var permission = permissions[i];
+
+                    if (!permission.isAuthorizeAllow()) {
+                        permissionOptions.userName = permission.getUserName();
+                        permissionOptions.formattedUserName = renderRoles(permission.getUserName());
+                        permissionOptions.readAllow = permission.isReadAllow();
+                        permissionOptions.readDeny = permission.isReadDeny();
+                        permissionOptions.writeAllow = permission.isWriteAllow();
+                        permissionOptions.writeDeny = permission.isWriteDeny();
+                        permissionOptions.deleteAllow = permission.isDeleteAllow();
+                        permissionOptions.deleteDeny = permission.isDeleteDeny();
+                        permissionOptions.notReadOnly = !(permission.getUserName() == "system/wso2.anonymous.role");
```

Comment:
Thanks for the comment. These are used only in two places and both of them are different cases. So making it a constant will not be needed.